docs/grammar/SolidityParser.g4

```diff
@@ -311,11 +311,35 @@ errorDefinition:
 	LParen (parameters+=errorParameter (Comma parameters+=errorParameter)*)? RParen
 	Semicolon;
 
+/**
+ * Operators that users are allowed to implement for some types with `using for`.
+ */
+userDefinableOperator:
+	BitAnd
+	| BitNot
+	| BitOr
+	| BitXor
+	| Sar
+	| Shl
+	| Add
+	| Div
+	| Exp
+	| Mod
+	| Mul
+	| Sub
+	| Equal
+	| GreaterThan
+	| GreaterThanOrEqual
+	| LessThan
+	| LessThanOrEqual
+	| NotEqual
+	| Not;
```

Comment:
`!` currently returns the same type it takes. I actually found it a bit awkward recently when I wanted to implement it for a type based on `uint`. I wonder if it wouldn't make more sense for it to return `bool` instead.

Reply:
Yes, it should be ``bool``. ``~`` is enough for a negationy thing within the type.

Reply:
Changed to `bool`.